pybamm/simulation.py

```diff
@@ -768,12 +769,19 @@ def solve(
 
                     logs["termination"] = step_solution.termination
                     # Only allow events specified by experiment
-                    if not (
+                    if (
+                        step_termination == "final time"
+                        and step.uses_default_duration is True
```

Comment:
We can probably skip the "is True" part here since it is a bool in the definition